fieldsolver/ldz_magnetic_field.cpp

```diff
@@ -191,8 +191,17 @@ void propagateSysBoundaryMagneticField(
       bGrid = perBDt2Grid.get(i,j,k);
    }
    cuint sysBoundaryFlag = technicalGrid.get(i,j,k)->sysBoundaryFlag;
+   cuint sysBoundaryLayer = technicalGrid.get(i,j,k)->sysBoundaryLayer;
+   
    for (uint component = 0; component < 3; component++) {
-      bGrid->at(fsgrids::bfield::PERBX + component) = sysBoundaries.getSysBoundary(sysBoundaryFlag)->fieldSolverBoundaryCondMagneticField(perBGrid, perBDt2Grid, EGrid, EDt2Grid, technicalGrid, i, j, k, dt, RKCase, component);
+      cint neigh_i=i + ((component==0)?-1:0);
+      cint neigh_j=j + ((component==1)?-1:0);
+      cint neigh_k=k + ((component==2)?-1:0);
+      cuint neighborSysBoundaryFlag = technicalGrid.get(neigh_i, neigh_j, neigh_k)->sysBoundaryFlag;
+      
+      if (neighborSysBoundaryFlag != sysboundarytype::NOT_SYSBOUNDARY) { // Complement to propagateMagneticFieldSimple main loop
```

Comment:
Should this be also != sysboundarytype::DO_NOT_COMPUTE ?

Reply:
Yep you also caught the first bug here. Line 200 segfaults at (i,j,k)=(0,0,0).

Reply:
Hmm, darn. I guess we'll need to have a special check to see if i, j, or k is zero then.

Reply:
My last commit solves this.

Reply:
5836b20 that is.

Reply:
Not really, as for L2 it just directly calls the ionosphere Bfield boundary condition - and as I listed below, that needs to access the -1 and +1 neighbours anyway.

Reply:
Right, I'm adding that one below too. Interestingly it didn't segfault there... :^)